recipe/build.sh

```diff
@@ -1,132 +1,137 @@
-  -DPROTOBUF_UPDATE_FILES=ON  `# should be used if using protobuf`        \
-  -DPYTHON_DEFAULT_EXECUTABLE=${PREFIX}/bin/python                        \
-  -DWITH_1394=OFF                                                         \
-  -DWITH_CUDA=OFF                                                         \
-  -DWITH_GTK=OFF                                                          \
-  -DWITH_ITT=OFF                                                          \
-  -DWITH_JASPER=OFF                                                       \
-  -DWITH_LAPACK=OFF                                                       \
-  -DWITH_MATLAB=OFF                                                       \
-  -DWITH_OPENCL=OFF                                                       \
-  -DWITH_OPENCLAMDBLAS=OFF                                                \
-  -DWITH_OPENCLAMDFFT=OFF                                                 \
-  -DWITH_OPENNI=OFF                                                       \
-  -DWITH_TESSERACT=OFF                                                    \
-  -DWITH_VA=OFF                                                           \
-  -DWITH_VA_INTEL=OFF                                                     \
-  -DWITH_VTK=OFF                                                          \
-  -DBUILD_opencv_python2=OFF
 
+export PKG_CONFIG_LIBDIR=$PREFIX/lib
 
-if [[ ! $? ]]; then
-  echo "configure failed with $?"
-  exit 1
+IS_PYPY=$(${PYTHON} -c "import platform; print(int(platform.python_implementation() == 'PyPy'))")
+
+LIB_PYTHON="${PREFIX}/lib/libpython${PY_VER}${SHLIB_EXT}"
+if [[ ${IS_PYPY} == "1" ]]; then
+    INC_PYTHON="$PREFIX/include/pypy${PY_VER}"
+else
+    INC_PYTHON="$PREFIX/include/python${PY_VER}"
 fi
 
-cmake --build .
+# FFMPEG building requires pkgconfig
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$PREFIX/lib/pkgconfig
+
+mkdir -p build
+cd build
+
+cmake -LAH -G "Ninja"                                                     \
+    ${CMAKE_ARGS}                                                         \
+    -DCMAKE_BUILD_TYPE="Release"                                          \
+    -DCMAKE_PREFIX_PATH=${PREFIX}                                         \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX}                                      \
+    -DCMAKE_INSTALL_LIBDIR="lib"                                          \
+    -DOPENCV_DOWNLOAD_TRIES=1\;2\;3\;4\;5                                 \
+    -DOPENCV_DOWNLOAD_PARAMS=INACTIVITY_TIMEOUT\;30\;TIMEOUT\;180\;SHOW_PROGRESS \
+    -DOPENCV_GENERATE_PKGCONFIG=ON                                        \
+    -DENABLE_CONFIG_VERIFICATION=ON                                       \
+    -DENABLE_PRECOMPILED_HEADERS=OFF                                      \
+    $OPENMP                                                               \
+    -DWITH_LAPACK=0                                                       \
+    -DCMAKE_CXX_STANDARD=17                                               \
```

Comment:
I went over briefly, but some flags were dropped, like `-DWITH_WEBP=1` and `-DBUILD_WEBP=0`.
It would be good to follow the link you put: https://github.com/opencv/opencv/blob/4.10.0/doc/tutorials/introduction/config_reference/config_reference.markdown#png-jpeg-tiff-webp-support

Reply:
Thanks. I revisited the link the compare it with a recipe. I made changes.